### PR TITLE
fix: Remove duplicate citations and add test to prevent it going forward

### LINKED ISSUE #4013: Add test to ensure avoid duplicate citations with different ids
Some datasets/models/benchamrks could have same paper in citation, but different ids which leads to duplications in paper

_Originally posted by @KennethEnevoldsen in https://github.com/embeddings-benchmark/mteb/pull/4012#pullrequestreview-3722397200_

### PR DESCRIPTION
**Pull request by omkar-334**
Fixes #4013   
Screenshot 2026-02-02 at 12 05 53 PM

### CODE REVIEW

**Comment by omkar-334** [comment]
cc @Samoed @KennethEnevoldsen  
found 9 duplicates, but 1 is false positive

File: mteb/get_tasks.py [hunk 1]
```diff
@@ -64,6 +66,100 @@ def _create_similar_tasks(tasks: Iterable[type[AbsTask]]) -> dict[str, list[str]
...
+                entries.append((citation_id, title))
+    return entries
+
+
+def _normalize_title_for_comparison(title: str) -> str:
+    # lowercase + collapse spaces so "CREMA-D" and "Crema-d" match
+    return " ".join(title.lower().strip().split())
+
+
+# titles that are just venue names (e.g. "Proceedings of LREC") — skip so we don't flag different papers from same conf
```
Comment (Samoed) [inline mteb/get_tasks.py]: I think better to not skip, because papers should have meaningful citations

File: mteb/get_tasks.py [hunk 2]
```diff
@@ -64,6 +66,100 @@ def _create_similar_tasks(tasks: Iterable[type[AbsTask]]) -> dict[str, list[str]
...
+    for norm_title, items in by_title.items():
+        if _is_venue_only_title(norm_title):
+            continue
+        id_to_raw = {cid: raw for _, cid, raw in items}
+        if len(id_to_raw) < 2:
+            continue
+        unique_ids = sorted(id_to_raw)
+        task_name = items[0][0]
+        for id1, id2 in combinations(unique_ids, 2):
+            duplicates.append(
```
Comment (Samoed) [inline mteb/get_tasks.py]: I think you can save also both tasknames

File: mteb/get_tasks.py [hunk 3]
```diff
@@ -64,6 +66,100 @@ def _create_similar_tasks(tasks: Iterable[type[AbsTask]]) -> dict[str, list[str]
...
+    if not normalized_title or len(normalized_title) < 20:
+        return True
+    lower = normalized_title.lower()
+    return any(lower.startswith(p) for p in _VENUE_ONLY_TITLE_PREFIXES) or lower in (
+        "acl",
+        "trec",
+    )
+
+
+def get_duplicate_citations() -> list[tuple[str, str, str, str, str]]:
```
Comment (Samoed) [inline mteb/get_tasks.py]: You can also try to use `difflib.get_close_matches` to find similar titles

File: mteb/get_tasks.py
Comment (Samoed) [inline mteb/get_tasks.py:1]: Can you move your changes to the test file?

File: mteb/get_tasks.py (hunk 1 above)
Comment (Samoed) [reply mteb/get_tasks.py]: It looks like the regex is handling this a bit incorrectly. For example, for DiaBla it picks up `booktitle` instead of `title`. I fixed this by updating the regex to match `\btitle`
```
        bibtex_citation=r"""
@inproceedings{gonzalez2019diabla,
  author = {González, Matilde and García, Clara and Sánchez, Lucía},
  booktitle = {Proceedings of the 12th Language Resources and Evaluation Conference},
  pages = {4192--4198},
  title = {DiaBLa: A Corpus of Bilingual Spontaneous Written Dialogues for Machine Translation},
  year = {2019},
}
""",

```

**Comment by omkar-334** [comment]
If we include comparing popular venues as well, I think we'll get lot more false positives because the Venue Names can match more easily. 
What do you think? @Samoed 
Example -
```
E         --- Duplicate 23: TV2Nordretrieval / JSTS ---
E           id1 = 'flansmose-mikkelsen-etal-2022-ddisco' (task: TV2Nordretrieval)
E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
E           id2 = 'kurihara-etal-2022-jglue' (task: JSTS)
E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
E         
E         --- Duplicate 24: TV2Nordretrieval / NaijaSenti ---
E           id1 = 'flansmose-mikkelsen-etal-2022-ddisco' (task: TV2Nordretrieval)
E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
E           id2 = 'muhammad-etal-2022-naijasenti' (task: NaijaSenti)
E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
```

**Comment by Samoed** [comment]
Can you also add to test also models and benchmarks?

**Comment by Samoed** [comment]
> If we include comparing popular venues as well, I think we'll get lot more false positives because the Venue Names can match more easily. What do you think? @Samoed Example -
> 
> ```
> E         --- Duplicate 23: TV2Nordretrieval / JSTS ---
> E           id1 = 'flansmose-mikkelsen-etal-2022-ddisco' (task: TV2Nordretrieval)
> E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
> E           id2 = 'kurihara-etal-2022-jglue' (task: JSTS)
> E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
> E         
> E         --- Duplicate 24: TV2Nordretrieval / NaijaSenti ---
> E           id1 = 'flansmose-mikkelsen-etal-2022-ddisco' (task: TV2Nordretrieval)
> E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
> E           id2 = 'muhammad-etal-2022-naijasenti' (task: NaijaSenti)
> E               title: Proceedings of the Thirteenth Language Resources and Evaluation Conference
> ```



@omkar-334 I fixed in https://github.com/embeddings-benchmark/mteb/pull/4032#discussion_r2753556358

**Comment by omkar-334** [comment]
> Can you also add to test also models and benchmarks?

sure, will do

**Comment by Samoed** [comment]
I added models, benchmarks and search for similar titles

**Comment by KennethEnevoldsen** [comment]
@Samoed I believe this is good to merge
